### DISCUSSION TITLE
Nextjs support

### DISCUSSION THREAD

**Discussion by davut** [ecadlabs/taquito#1395] [Q&A]
Hi,

Does taquito have Nextjs support? I was getting a lot of localstorage not found errors. I know what this error is but if I am going to get lots of error in my projects I would not use it.

Did someone use it in their Nextjs project? How it went?

Would love some help thanks.

**Comment by jonah-elbaz-volvo** [comment]
Ive used https://www.npmjs.com/package/patch-package
And included this diff.
Seems to work.
Basically airgap uses local storage, but thankfully they centralized all its use in one file so you can define it even without window access.

```
diff --git a/node_modules/@airgap/beacon-core/dist/cjs/storage/LocalStorage.js b/node_modules/@airgap/beacon-core/dist/cjs/storage/LocalStorage.js
index 5270f20..a4ca8a2 100644
--- a/node_modules/@airgap/beacon-core/dist/cjs/storage/LocalStorage.js
+++ b/node_modules/@airgap/beacon-core/dist/cjs/storage/LocalStorage.js
@@ -43,6 +43,14 @@ var beacon_types_1 = require("@airgap/beacon-types");
  *
  * A storage that can be used in the browser
  */
+let localStorage = typeof window !== "undefined" ? window.localStorage : {
+    getItem(key) {
+        return null
+    },
+    setItem(key, value) {
+        return null
+    }
+}
 var LocalStorage = /** @class */ (function () {
     function LocalStorage(prefix) {
         this.prefix = prefix;
```
